### ISSUE TITLE
Bug: Channels not loading in indigenous for ios

### ISSUE THREAD

**Issue by jackjamieson2** [jackjamieson2/yarns-microsub-server#8]
See https://github.com/EdwardHinkle/indigenous-ios/issues/224

**Comment by EdwardHinkle** [comment]
Working on getting this installed on my Wordpress testing server. I downloaded the source, and put it into the plugins directory and renamed the folder "yarns-microsub-server". I see it in the plugins listing but when I click to activate it presents the following error: `Warning: require_once(/home/ubuntu/production/wordpress.abode.pub/wp-content/plugins/yarns-microsub-server/vendor/autoload.php): failed to open stream: No such file or directory in /home/ubuntu/production/wordpress.abode.pub/wp-content/plugins/yarns-microsub-server/yarns-microsub.php on line 15
`

**Comment by EdwardHinkle** [comment]
Is there something else I need to do in order to be able to activate it?

**Comment by dshanske** [comment]
Are you using composer?

**Comment by jackjamieson2** [comment]
@EdwardHinkle Should work now, I updated the repo to make sure the necessary vendor files are included. Sorry I missed that earlier

There's no front-end at all, so you'll need to add channels, follow feeds, etc fully through the reader.  

Lacks the ability to delete posts, re-order channels, delete channels, and maybe a few more things, but the basic ability to follow feeds should be reasonably functional

**Comment by EdwardHinkle** [comment]
Sounds good. That's enough functionality for me to be able to troubleshoot what's going wrong with Indigenous. Thanks!

**Comment by EdwardHinkle** [comment]
Hmmm so I downloaded the master branch into my plugins directory as "yarns-microsub-server", once inside I ran `composer install`.

I got an error:
```
Update failed (The .git directory is missing from vendor/parse-this, see https://getcomposer.org/commit-deps for more information)
    Would you like to try reinstalling the package instead [yes]? yes
```

I selected yes and it Removed and reinstalled parse-this and completed.

Then, I opened up the Wordpress admin panel and activated the plugin. My Wordpress test instance now has the PHP WSOD (White Screen of Death)

**Comment by EdwardHinkle** [comment]
When I deleted the yarns-microsub-server plugin, my Wordpress started working again

**Comment by jackjamieson2** [comment]
Argh, sorry!  I changed how parse-this is added and must have made an
error. I’m away from my computer today but I can fix this tomorrow.

**Comment by EdwardHinkle** [comment]
No worries :) Just wanted to make sure I wasn't doing something wrong on my end

**Comment by dshanske** [comment]
I am having that problem too. I think parse this should be listed as a dev dependency, not a live one.

**Comment by jackjamieson2** [comment]
@EdwardHinkle I just fixed the dependencies and it seems to be working now - thanks for letting me know!

**Comment by EdwardHinkle** [comment]
Awesome, I’ll try and check it out here today or tomorrow. Hopefully I can figure out what’s going wrong with Indigenous 🤞

**Comment by jackjamieson2** [comment]
Thanks @EdwardHinkle! There's a high chance it's a problem on my end, but I'm stumped so glad for a second set of eyes

**Comment by EdwardHinkle** [comment]
Oh great! Last time I tried I didn’t think the dependencies were included. Great to have those included now! 👍

**Comment by jackjamieson2** [comment]
@EdwardHinkle, Did you ever get a chance to check this out?  Or is there anything I can do to help debug?

If nothing else, if you implement https://github.com/EdwardHinkle/indigenous-ios/issues/241 I'll check for error messages and (a) fix them in Yarns or (b) send you a detailed issue if I can't address the problem on my end.

**Comment by jackjamieson2** [comment]
Moved this from the Beta release milestone to a 1.0 milestone, since I'm still stumped on this problem.

Some thinking aloud
- Since Aperture works with Indigenous for iOS, take a close look at aperture's responses, and modify mine to match.  (In general, I think my code can only improve by following @aaronpk's example).
- Maybe my response is missing some necessary headers or something like that — in which case I could improve things by creating a function to generate a more complete JSON response, instead of just using `return ${json}`

**Comment by jackjamieson2** [comment]
see https://chat.indieweb.org/wordpress/2019-01-07#t1546888448260600

@dshanske mentioned that Indigenous for Android also doesn't work with Yarns. Hopefully they're having the same issue so I can fix both in one swoop.

**Comment by jackjamieson2** [comment]
This is fixed in ca7472959c0a9e1ad1f3c31d971e218c078e393f

The problem was that Yarns' endpoint responses just used `return $json` but I should have used `WP_REST_Response`
